pkgs/native_assets_cli/lib/src/model/hook_config.dart

```diff
@@ -250,73 +250,72 @@ abstract class HookConfigImpl implements HookConfig {
   static Uri parsePackageRoot(Uri baseUri, Map<String, Object?> config) =>
       config.path(packageRootConfigKey, baseUri: baseUri, mustExist: true);
 
-  static BuildModeImpl? parseBuildMode(
-      Map<String, Object?> config, bool dryRun) {
+  static BuildMode? parseBuildMode(Map<String, Object?> config, bool dryRun) {
     if (dryRun) {
-      _throwIfNotNullInDryRun<String>(config, BuildModeImpl.configKey);
+      _throwIfNotNullInDryRun<String>(config, _buildModeConfigKey);
       return null;
     } else {
-      return BuildModeImpl.fromString(
+      return BuildMode.fromString(
         config.string(
-          BuildModeImpl.configKey,
-          validValues: BuildModeImpl.values.map((e) => '$e'),
+          _buildModeConfigKey,
+          validValues: BuildMode.values.map((e) => '$e'),
         ),
       );
     }
   }
 
-  static LinkModePreferenceImpl parseLinkModePreference(
+  static LinkModePreference parseLinkModePreference(
           Map<String, Object?> config) =>
-      LinkModePreferenceImpl.fromString(
+      LinkModePreference.fromString(
         config.string(
-          LinkModePreferenceImpl.configKey,
-          validValues: LinkModePreferenceImpl.values.map((e) => e.toString()),
+          _linkModePreferenceConfigKey,
+          validValues: LinkModePreference.values.map((e) => e.toString()),
```

Comment:
This line uses `(e) => e.toString()`, line 261 uses `(e) => '$e'`.
Choose one and be consistent.

Consider making it a static helper function, seems emminently sheareable as:
```dart
String _toString(Object? o) => o.toString();
```

Reply:
I'll go with the shorter string interpolation one. Given the code is so small, I'll do it inline. (We really should allow compilers to make such closures (closures that don't capture anything) static automatically and relax identity guarantees ...)